public/lessons/greedy/intro.mdx

```diff
@@ -0,0 +1,19 @@
+# Introduction to Greedy Algorithms in Competitive Programming
+
+## What are Greedy Algorithms?
+Competitive programming often involves solving complex problems under tight time constraints. Greedy algorithms are a fundamental technique used to design efficient and effective solutions in this context. A greedy algorithm makes locally optimal choices at each step with the hope that these choices will lead to a globally optimal solution.
+
+The key idea behind greedy algorithms is to make the best possible decision at each stage of the problem, without reconsidering or undoing previous choices. Unlike dynamic programming, which stores and reuses solutions to subproblems, greedy algorithms focus on making immediate decisions to maximize some objective function.
+
+## Characteristics of Greedy Algorithms
+Greedy algorithms exhibit several common characteristics:
+
+### 1. Greedy Choice Property
+At each step, a greedy algorithm makes the choice that appears to be the best at the moment. This choice is determined based on a specific criterion or heuristic.
```

Comment:
Maybe state "This choice is determined based on a specific criterion, such as a certain element having a greater value." I would recommend avoiding the word heuristic as that usually refers to an imperfect greedy approach.